packages/api/src/auth/verifiers/jwtVerifier.ts

```diff
@@ -0,0 +1,90 @@
+import jwt from 'jsonwebtoken'
+
+import {
+  VerifyOptions,
+  WebhookSignError,
+  WebhookVerificationError,
+  DEFAULT_WEBHOOK_SECRET,
+} from './index'
+import type { WebhookVerifier } from './index'
+
+export type JwtVerifier = WebhookVerifier
+
+/**
+ *
+ * createSignature
+ *
+ */
+const createSignature = ({
+  payload,
+  secret = DEFAULT_WEBHOOK_SECRET,
+  options,
+}: {
+  payload: string | Record<string, unknown>
+  secret: string
+  options: VerifyOptions
+}): string => {
+  try {
+    const signOptions = options.issuer ? { issuer: options.issuer } : undefined
+
+    return jwt.sign(payload, secret, { ...signOptions })
+  } catch (error) {
+    throw new WebhookSignError(error.message)
+  }
+}
+
+/**
+ *
+ * verifySignature
+ *
+ */
+export const verifySignature = ({
+  payload,
+  secret = DEFAULT_WEBHOOK_SECRET,
+  signature,
+  options,
+}: {
+  payload: string | Record<string, unknown>
+  secret: string
+  signature: string
+  options: VerifyOptions
+}): boolean => {
+  try {
+    if (payload === undefined || payload?.length === 0) {
+      console.warn('Missing payload')
+    }
+
+    const decoded = jwt.verify(signature, secret) as Record<string, unknown>
+
+    if (decoded['iss'] && decoded['iss'] !== options.issuer) {
```

Comment:
I think you can pass `issuer` as an option to jwt.verify, instead of doing this manually in a second step.

We also need a check for if issuer is specified, as it may not always be verified

Reply:
Something like this:

```ts
     const decoded = jwt.verify(signature, secret, {issuer: options.issuer}) as Record<string, unknown>
```
Not 100% sure if the jwt library will accept undefined and carry on if issuer isn't specified in the options